challenge-10/challenge-10.js

```diff
@@ -1,113 +1,179 @@
+    propriedade, usando os valores passados por parâmetro.
+    */
+    var operation = {
+        '+': function( numberOne, numberTwo ) {
+            return numberOne + numberTwo;
+        },
+        '-': function( numberOne, numberTwo ) {
+            return numberOne - numberTwo;
+        },
+        '*': function( numberOne, numberTwo ) {
+            return numberOne * numberTwo;
+        },
+        '/': function( numberOne, numberTwo ) {
+            return numberOne / numberTwo;
+        },
+        '%': function( numberOne, numberTwo ) {
+            return numberOne % numberTwo;
+        }
+    };
+
+    /*
+    Crie uma função chamada `isOperatorValid`, que receberá um operador por
+    parâmetro.
+    - Essa função será responsável por verificar se o operador passado por
+    parâmetro a ela é válido, ou seja, se ele é igual a '+', '-', '*', '/' ou
+    '%'.
+    - Se for igual a qualquer um desses, ela deverá retornar "true".
+    Caso contrário, "false".
+    - O desafio é fazer o retorno sem usar "if" ou "switch".
+    */
+    function isOperatorValid( operator ) {
+        return !!operation[operator];
+    }
+
+    /*
+    Agora vamos criar a calculadora.
+    - Crie uma função chamada `calculator`, que receberá como parâmetro um
+    operador;
+    - Se o operador não for válido, a função deve retornar "false";
+    - Se o operador for válido, retornar uma segunda função que receberá dois
+    parâmetros;
+    - Se algum dos parâmetros não for um número, retornar "false";
+    - Senão, retornar o método do objeto "operation" criado acima, baseado no
+    operador passado para a função "calculator", e passando para esse método
+    os dois parâmetros da função de retorno de "calculator".
+    */
+    function calculator( operator ) {
+        if( isOperatorValid( operator ) === false ) {
+            return false;
+        }
+
+        return function( numberOne, numberTwo ) {
+            if( typeof numberOne !== 'number' || typeof numberTwo !== 'number') {
+                return false;
+            }
+
+            return operation[operator]( numberOne, numberTwo );
+        }
+    }
+
+    /*
+    Crie uma função chamada "showOperationMessage" que recebe três parâmetros:
+    - o operador, o primeiro número e o segundo número. O retorno da função
+    deve ser a frase:
+    'A operação [NUMBER1] [OPERATOR] [NUMBER2] =';
+    Essa função mostrará a mensagem da operação que criaremos mais abaixo.
+    */
+    function showOperationMessage( operator, numberOne, numberTwo ) {
+        return 'A operação ' + numberOne + ' ' + operator + ' ' + numberTwo + ' = ';
+    }
+
+    /*
+    Crie uma função chamada "showErrorMessage" que recebe um parâmetro: o
+    operador da operação cálculo, quando a operação não for válida.
+    Essa função deverá retornar a frase:
+    'Operação "[OPERATOR]" não permitida!'
+    */
+    function showErrorMessage( operator ) {
+         if( isOperatorValid( operator ) === false) {
```

Comment:
-1
Essa função não precisa verificar nada, pois ela só será chamada quando o operador não for válido ;)